packages/forms/frontend/sirius-components-forms/src/propertysections/DateTimeWidgetPropertySection.tsx

```diff
@@ -0,0 +1,255 @@
+ *******************************************************************************/
+import { MutationResult, gql, useMutation } from '@apollo/client';
+import { getCSSColor, useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import TextField from '@material-ui/core/TextField';
+import { Theme, makeStyles } from '@material-ui/core/styles';
+import { FocusEvent, useEffect, useState } from 'react';
+import { PropertySectionComponent, PropertySectionComponentProps } from '../form/Form.types';
+import { GQLDateTime } from '../form/FormEventFragments.types';
+import {
+  DateTimeStyleProps,
+  GQLEditDateTimeInput,
+  GQLEditDateTimeMutationData,
+  GQLEditDateTimeMutationVariables,
+  GQLPayload,
+  GQLUpdateWidgetFocusInput,
+  GQLUpdateWidgetFocusMutationData,
+  GQLUpdateWidgetFocusMutationVariables,
+} from './DateTimeWidgetPropertySection.types';
+import { PropertySectionLabel } from './PropertySectionLabel';
+
+const useStyle = makeStyles<Theme, DateTimeStyleProps>((theme) => ({
+  style: {
+    backgroundColor: ({ backgroundColor }) => (backgroundColor ? getCSSColor(backgroundColor, theme) : null),
+    color: ({ foregroundColor }) => (foregroundColor ? getCSSColor(foregroundColor, theme) : null),
+    fontStyle: ({ italic }) => (italic ? 'italic' : null),
+    fontWeight: ({ bold }) => (bold ? 'bold' : null),
+  },
+  textfield: {
+    marginTop: theme.spacing(0.5),
+    marginBottom: theme.spacing(0.5),
+  },
+}));
+
+export const updateWidgetFocusMutation = gql`
+  mutation updateWidgetFocus($input: UpdateWidgetFocusInput!) {
+    updateWidgetFocus(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        messages {
+          body
+          level
+        }
+      }
+    }
+  }
+`;
+
+export const editDateTimeMutation = gql`
+  mutation editDateTime($input: EditDateTimeInput!) {
+    editDateTime(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        messages {
+          body
+          level
+        }
+      }
+      ... on SuccessPayload {
+        messages {
+          body
+          level
+        }
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload): payload is GQLPayload => payload.__typename === 'ErrorPayload';
+const isSuccessPayload = (payload): payload is GQLPayload => payload.__typename === 'SuccessPayload';
+
+function useErrorReporting<T>(
+  result: MutationResult<T>,
+  extractPayload: (data: T | null) => GQLPayload | undefined,
+  addErrorMessage: (message: string) => any,
+  addMessages: (messages: any) => any
+) {
+  useEffect(() => {
+    const { loading, data, error } = result;
+    if (!loading) {
+      if (error) {
+        addErrorMessage(error.message);
+      }
+      const payload: GQLPayload | undefined = extractPayload(data || null);
+      if (payload && (isSuccessPayload(payload) || isErrorPayload(payload))) {
+        const { messages } = payload;
+        addMessages(messages);
+      }
+    }
+  }, [result.loading, result.data, result.error]);
+}
```

Comment:
yes but not in the PR. We must define the scope of this change because currently `useErrorReporting` pattern is used in portal, deck, gantt and DateTimeWidget. Ideally, we should use it everywhere it is needed. It implies lots of changes including GQLPayloadXxx that are define everywhere.

Reply:
We could simplify `useErrorReporting` by using `useMultiToast` inside. Then we would have only result and extractPayload parameters.
We could even force all `GQLxxxMutationData` to have a payload attribute and then only the result parameter would be necessary.

Reply:
> yes but not in the PR. We must define the scope of this change because currently useErrorReporting pattern is used in portal, deck, gantt and DateTimeWidget.

Yes in this PR. I don't know why this kind of situation happens again and again where people try to improve just their own piece of code. This code should either be removed or in sirius-components-core and that's the same for its other occurences.

>  It implies lots of changes including GQLPayloadXxx that are define everywhere.

I'm not asking for all pieces of code where errors are treated to be updated, I'm just asking for this specific piece of code, those 20 lines of code, to be accessible by everybody by being in sirius-components-core. And if they are moved there, it should be very easy to update both the portal and gantt part too instead of copying and pasting stuff.

Reply:
done